src/http_tracker_connection.cpp

```diff
@@ -117,7 +117,7 @@ namespace libtorrent
 		url += "info_hash=";
 		url += escape_string((const char*)&tracker_req().info_hash[0], 20);
 
-		if (tracker_req().kind == tracker_request::announce_request)
+		if (0 == (tracker_req().kind & tracker_request::scrape_request))
```

Comment:
Somehow your message didn't make it here... probably as a result of forced push...
> wouldn't a more reliable way (i.e. less likely to introduce errors) to make the new i2p flag not be a flag, but just a part of the enumeration, and call it i2p_announce or something like that?

And add i2p_scrape as well and compare for this or that? I feel like having 2 independent flags announce/scrape and is_i2p would be the least error prone. Are there many places that use announce? Maybe renaming *announce_request* to *announce_request_dontuse* is an ok solution?